From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "emmanuel.kimmerlin@thomsonreuters.com"
 <emmanuel.kimmerlin@thomsonreuters.com>
Date: Mon, 18 Nov 2019 12:42:23 +0100
Subject: Expose a method for invalidating internal cache

Allow to customize Electron/Chromium cache location
See MR: https://github.com/electron/electron/pull/19693

diff --git a/base/path_service.cc b/base/path_service.cc
index d1c9bb1045a5d4cff7ded79ebced53ce4c2daa3c..01ef3fd68bd1683b973a9545639c3d2ad116fae3 100644
--- a/base/path_service.cc
+++ b/base/path_service.cc
@@ -344,6 +344,15 @@ void PathService::RegisterProvider(ProviderFunc func, int key_start,
   path_data->providers = p;
 }
 
+// static
+void PathService::InvalidateCache() {
+  PathData * path_data = GetPathData();
+  DCHECK(path_data);
+
+  AutoLock scoped_lock(path_data->lock);
+  path_data->cache.clear();
+}
+
 // static
 void PathService::DisableCache() {
   PathData* path_data = GetPathData();
diff --git a/base/path_service.h b/base/path_service.h
index c0cb678df47647a4bb0866d0b82c80e44f0dc021..a22505ca0e652d51faa2d0075b4fe3ed167e74b9 100644
--- a/base/path_service.h
+++ b/base/path_service.h
@@ -76,6 +76,9 @@ class BASE_EXPORT PathService {
   // Disable internal cache.
   static void DisableCache();
 
+  // Clear cache
+  static void InvalidateCache();
+
  private:
   friend class ScopedPathOverride;
   FRIEND_TEST_ALL_PREFIXES(PathServiceTest, RemoveOverride);
